[prometheus-blackbox-exporter] Linting fails due to use of deprecated apiVersion rbac.authorization.k8s.io/v1beta1

**Describe the bug**

When running `helm lint` for the `prometheus-blackbox-exporter` chart `4.3.1` it fails with the following errors:

```
  ==> Linting /tmp/750049035/monitoring/prometheus-blackbox-exporter/helm-charts-stable/prometheus-blackbox-exporter/4.3.0/prometheus-blackbox-exporter
  [INFO] Chart.yaml: icon is recommended
  [ERROR] templates/role.yaml: the kind "rbac.authorization.k8s.io/v1beta1 Role" is deprecated in favor of "rbac.authorization.k8s.io/v1 Role"
  [ERROR] templates/rolebinding.yaml: the kind "rbac.authorization.k8s.io/v1beta1 RoleBinding" is deprecated in favor of "rbac.authorization.k8s.io/v1 RoleBinding"
  Error: 1 chart(s) linted, 1 chart(s) failed
```

**Version of Helm and Kubernetes**:

Helm `v3.3.1`
Kubernetes `v1.19.0`

**Which chart**:

prometheus-blackbox-exporter

**What happened**:

Fails linting due to a deprecated api version in `role.yaml` and `rolebinding.yaml`

**What you expected to happen**:

Linting to succeed

**How to reproduce it** (as minimally and precisely as possible):

* Clone the repository
* Navigate to directory `charts/prometheus-blackbox-exporter`
* Run Helm (3+) linting with `helm lint`

**Anything else we need to know**:

Nope

--- a/charts/prometheus-blackbox-exporter/Chart.yaml
+++ b/charts/prometheus-blackbox-exporter/Chart.yaml
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: Prometheus Blackbox Exporter
 name: prometheus-blackbox-exporter
-version: 4.3.1
+version: 4.3.2
 appVersion: 0.16.0
 home: https://github.com/prometheus/blackbox_exporter
 sources:

--- a/charts/prometheus-blackbox-exporter/templates/_helpers.tpl
+++ b/charts/prometheus-blackbox-exporter/templates/_helpers.tpl
@@ -40,3 +40,14 @@ Create the name of the service account to use
     {{ default "default" .Values.serviceAccount.name }}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Return the appropriate apiVersion for rbac.
+*/}}
+{{- define "rbac.apiVersion" -}}
+{{- if .Capabilities.APIVersions.Has "rbac.authorization.k8s.io/v1" }}
+{{- print "rbac.authorization.k8s.io/v1" -}}
+{{- else -}}
+{{- print "rbac.authorization.k8s.io/v1beta1" -}}
+{{- end -}}
+{{- end -}}

--- a/charts/prometheus-blackbox-exporter/templates/role.yaml
+++ b/charts/prometheus-blackbox-exporter/templates/role.yaml
@@ -1,5 +1,5 @@
 {{- if .Values.pspEnabled }}
-apiVersion: rbac.authorization.k8s.io/v1beta1
+apiVersion: rbac.authorization.k8s.io/v1
 kind: Role
 metadata:
   labels:

--- a/charts/prometheus-blackbox-exporter/templates/role.yaml
+++ b/charts/prometheus-blackbox-exporter/templates/role.yaml
@@ -1,5 +1,5 @@
 {{- if .Values.pspEnabled }}
-apiVersion: rbac.authorization.k8s.io/v1
+apiVersion: {{ template "rbac.apiVersion" . }}
 kind: Role
 metadata:
   labels:

--- a/charts/prometheus-blackbox-exporter/templates/rolebinding.yaml
+++ b/charts/prometheus-blackbox-exporter/templates/rolebinding.yaml
@@ -1,5 +1,5 @@
 {{- if .Values.pspEnabled }}
-apiVersion: rbac.authorization.k8s.io/v1beta1
+apiVersion: rbac.authorization.k8s.io/v1
 kind: RoleBinding
 metadata:
   labels:

--- a/charts/prometheus-blackbox-exporter/templates/rolebinding.yaml
+++ b/charts/prometheus-blackbox-exporter/templates/rolebinding.yaml
@@ -1,5 +1,5 @@
 {{- if .Values.pspEnabled }}
-apiVersion: rbac.authorization.k8s.io/v1
+apiVersion: {{ template "rbac.apiVersion" . }}
 kind: RoleBinding
 metadata:
   labels:

